platforms/cpuinfo_linux.go

```diff
@@ -0,0 +1,88 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package platforms
+
+import (
+	"bytes"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"golang.org/x/sys/unix"
+)
+
+// getMachineArch retrieves the machine architecture through system call
+// same result from uname -m running the terminal
+// This function is favored to the previous method of getting variant information from /proc/cpuinfo
+// because /proc/cpuinfo is not accurate in user-mode emulation environment
+func getMachineArch() (string, error) {
+	var uname unix.Utsname
+	err := unix.Uname(&uname)
+	if err != nil {
+		return "", err
+	}
+
+	arch := string(uname.Machine[:bytes.IndexByte(uname.Machine[:], 0)])
+
+	return arch, nil
+}
+
+// getCPUVariant returns cpu variant for ARM
+// return empty string in case of error
+func getCPUVariant() string {
+
+	if !isArmArch(runtime.GOARCH) {
+		log.L.Errorf("getCPUVariant for ARCH %s: %v", runtime.GOARCH, errdefs.ErrNotImplemented)
+	}
+
+	var variant string
+
+	arch, err := getMachineArch()
+	if err != nil {
+		log.L.WithError(err).Error("failure getting machine architecture")
+		return ""
+	}
+
+	arch = strings.ToLower(arch)
+
+	if arch == "aarch64" {
+		variant = "v8"
+	} else if len(arch) >= 5 {
+		//Expect the valid arch format here to be armvXx
+		switch arch[3:5] {
+		case "v8":
+			variant = "v8"
+		case "v7":
+			variant = "v7"
+		case "v6":
+			variant = "v6"
+		case "v5":
+			variant = "v5"
+		case "v4":
+			variant = "v4"
+		case "v3":
+			variant = "v3"
+		default:
+			variant = "unknown"
+		}
+	} else {
+		log.L.WithError(err).Error("getCPUVariant invalid arch")
+		return ""
```

Comment:
What is the difference between `""` and `"unknown"`?

Reply:
I interpreted it as return "" if errors occurred,  return "unknown" if the field is not understandable. 
Let me standardize it as "unknown" for all cases.